### ISSUE TITLE
dotnet run NREs on project whose dependencies were not restored

### ISSUE THREAD

**Issue by moozzyk** [dotnet/cli#941]
I cloned https://github.com/davidfowl/dotnetcli-aspnetcore and forgot to run restore:
```
PS C:\source\dotnetcli-aspnetcore\HelloMvc> dotnet run
Object reference not set to an instance of an object.
```

(Not sure if it is relevant - I did not have `DOTNET_HOME` set)

**Comment by blackdwarf** [comment]
@moozzyk this doesn't repro for me on the latest bits. Will close this, plese reopen if you find it still repros.

**Comment by moozzyk** [comment]
When I run `dotnet run` in a folder an application was published to:

```
PS C:\temp\publishout> dotnet -v run
Telemetry is: Enabled
Object reference not set to an instance of an object.
```

```
PS C:\temp\publishout> dotnet --info
.NET Command Line Tools (1.0.0-rc2-002655)

Product Information:
 Version:     1.0.0-rc2-002655
 Commit Sha:  3e96a05f9c

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.10586
 OS Platform: Windows
 RID:         win10-x64
```

**Comment by eerhardt** [comment]
I think this was resolved with #2980.  Can you retry with the latest Preview2 bits and close if this is fixed?

**Comment by piotrpMSFT** [comment]
I verified this is fixed in preview2.
